internal/repositories/user_repository.go

```diff
@@ -49,13 +56,21 @@ func (r userRepository) Get(ctx context.Context, userDto dto.User) (*ent.User, e
 		return nil, ErrUserNotFound
 	}
 	if err != nil {
-		return nil, fmt.Errorf("querying user: %w", err)
+		return nil, ErrDatabase
 	}
 
 	return userData, nil
 }
 
-func (r userRepository) CreateUser(ctx context.Context, userData dto.User) error {
+func (r userRepository) Create(ctx context.Context, userData dto.User) error {
```

Comment:
Using the user model instead of the DTO in the repository layer is better. The model is generated in the service (BL) layer and sent to the repository.